File: frontend/src/services/FileHandlerFactory.js
// Файл: frontend/src/services/filehandlers/FileHandlerFactory.js
import MarkdownHandler from './MarkdownHandler'
import JsonHandler     from './JsonHandler'
import ImageHandler    from './ImageHandler'
import GenericHandler  from './GenericHandler'

export default {
  /**
   * @param {string} ext — расширение, включая точку, например ".txt"
   * @returns {{ validate: function }}
   */
  create(ext) {
    switch (ext.toLowerCase()) {
      case '.md':    return MarkdownHandler
      case '.json':  return JsonHandler
      case '.png':
      case '.jpg':
      case '.jpeg':  return ImageHandler

      // Добавляем все остальные текстовые/документные форматы сюда:
      case '.txt':
      case '.pdf':
      case '.doc':
      case '.docx':
      case '.xls':
      case '.xlsx':
      case '.csv':
        return GenericHandler

      // По умолчанию — тоже Generic
      default:
        return GenericHandler
    }
  }
}

File: backend/services/filehandlers/FileHandlerFactory.cs
using System;
using System.Collections.Generic;

namespace Mirea.freelance.backend.services.filehandlers
{
    public static class FileHandlerFactory
    {
        // Словарь «расширение → функция-создатель обработчика»
        private static readonly Dictionary<string, Func<IFileHandler>> _handlers
            = new Dictionary<string, Func<IFileHandler>>(StringComparer.OrdinalIgnoreCase)
        {
            [".md"]   = () => new MarkdownHandler(),
            [".json"] = () => new JsonHandler(),
            [".png"]  = () => new ImageHandler(),
            [".jpg"]  = () => new ImageHandler(),
            [".jpeg"] = () => new ImageHandler(),
            // можно здесь же по умолчанию добавить:
            [".txt"]  = () => new TextHandler(),
            [".pdf"]  = () => new BinaryHandler(),
            [".docx"] = () => new BinaryHandler(),
            [".xlsx"] = () => new BinaryHandler(),
        };

        /// <summary>
        /// Регистрирует новый обработчик для указанного расширения.
        /// </summary>
        public static void Register(string extension, Func<IFileHandler> creator)
        {
            if (extension == null) throw new ArgumentNullException(nameof(extension));
            if (!extension.StartsWith(".")) extension = "." + extension;
            _handlers[extension] = creator;
        }

        /// <summary>
        /// Возвращает обработчик для данного расширения.
        /// Если расширение не зарегистрировано — бросает NotSupportedException.
        /// </summary>
        public static IFileHandler Create(string extension)
        {
            if (extension == null) throw new ArgumentNullException(nameof(extension));
            if (!extension.StartsWith(".")) extension = "." + extension;

            if (_handlers.TryGetValue(extension, out var creator))
                return creator();

            throw new NotSupportedException($"Unsupported file type: {extension}");
        }
    }
}
